src/thriveopedia/pages/ThriveopediaMuseumPage.cs

```diff
@@ -82,20 +85,45 @@ public override void OnThriveopediaOpened()
 
     private void RefreshIfDirty()
     {
-        // TODO: caching or something here would make this a lot more efficient as for people with a ton of fossils
-        // the Thriveopedia likely will start to lag a lot
         GD.Print("Refreshing the Thriveopedia museum page");
-        cardContainer.QueueFreeChildren();
+
+        // Add already present fossil names and cards to fossilMemory
+        var count = cardContainer.GetChildCount();
+        for (int i = 0; i < count; ++i)
+        {
+            var card = cardContainer.GetChild<MuseumCard>(i);
+            fossilMemory.Add(card.OriginalName, card);
```

Comment:
As per our style guide, fossilMemory should be the *primary* data storage and the data should always go:

fossilMemory -> Godot Nodes

instead of this opposite: from nodes to our collection.